src/logic/safe/transactions/gas.ts

```diff
@@ -92,6 +93,141 @@ export const estimateTxGasCosts = async (
+
+      if (sameString(result, EMPTY_DATA)) {
+        // We throw if there is an error
+        throw new Error('There was an error retrieving the gas estimation')
+      }
+      return {
+        gasEstimationResponse: result,
+        isGethNode: true,
+      }
+    })
+    .catch((error) => {
+      // Non geth nodes
+      // When we calculate the gas for Non GETH nodes we will always receive the response as error
+      // In that case we check if there is empty data (which means an error)
+      // Or if there is a valid estimation amount
+      const estimationData = getNonGethErrorDataResult(error.message)
+
+      if (!estimationData || sameString(estimationData, EMPTY_DATA)) {
+        throw error
+      }
+
+      return {
+        gasEstimationResponse: estimationData,
+        isGethNode: false,
+      }
+    })
+}
+
+const calculateGasForGethNodes = async (
+  additionalGasBatches: number[],
+  safeAddress: string,
+  estimateData: string,
+  txGasEstimation: number,
+  dataGasEstimation: number,
+): Promise<number> => {
+  const web3 = getWeb3()
+  const batch = new web3.BatchRequest()
+  const estimationRequests = additionalGasBatches.map(
+    (additionalGas) =>
+      new Promise((resolve) => {
+        // there are no type definitions for .request, so for now ts-ignore is there
+        // Issue link: https://github.com/ethereum/web3.js/issues/3144
+        // eslint-disable-next-line
+        // @ts-ignore
+        const request = web3.eth.call.request(
+          {
+            to: safeAddress,
+            from: safeAddress,
+            data: estimateData,
+            gasPrice: 0,
+            gasLimit: txGasEstimation + dataGasEstimation + additionalGas,
+          },
+          (error, res) => {
+            // res.data check is for OpenEthereum/Parity revert messages format
+            const isOpenEthereumRevertMsg = res && typeof res.data === 'string'
+            const isEstimationSuccessful =
+              !error &&
+              ((typeof res === 'string' && res !== EMPTY_DATA) ||
+                (isOpenEthereumRevertMsg && res.data.slice(9) !== EMPTY_DATA))
+
+            resolve({
+              success: isEstimationSuccessful,
+              estimation: txGasEstimation + additionalGas,
+            })
+          },
+        )
+
+        batch.add(request)
+      }),
+  )
+  batch.execute()
+
+  const estimationResponses = await Promise.all(estimationRequests)
+  const firstSuccessfulRequest: any = estimationResponses.find((res: any) => res.success)
+
+  if (firstSuccessfulRequest) {
+    return firstSuccessfulRequest.estimation
+  }
+  return 0
+}
+
+const getNonGethErrorDataResult = (errorMessage: string): string | undefined => {
+  // Extracts JSON object from the error message
+  const [, ...error] = errorMessage.split('\n')
+  const errorAsJSON = JSON.parse(error.join(''))
+
+  if (errorAsJSON?.data) {
+    const [, dataResult] = errorAsJSON.data.split('Reverted ')
```

Comment:
This only works for OpenEthereum/Parity ... but not for Nethermind

Reply:
Yes you are right, I check this again and replaced `non-geth` comments/checks with `OpenEthereum` because it's the one failing. On `Geth` and `Nethermind` nodes, the initial implementation it's working (nethermind returns the result the same way as get, not as an error like openEthereum).